app/controllers/time_clock_periods_controller.rb

```diff
@@ -0,0 +1,64 @@
+class TimeClockPeriodsController < InternalController
+  before_action :set_time_clock_period, only: %i[ show edit update destroy ]
+
+  # GET /time_clock_periods
+  def index
+    @q = policy_scope(TimeClockPeriod).ransack(params[:q])
+    @time_clock_periods = @q.result(distinct: true).page(params[:page])
+  end
+
+  # GET /time_clock_periods/1
+  def show
+    @q = policy_scope(TimeClockPunch).where(time_clock_period: @time_clock_period).ransack(params[:q])
+    @time_clock_punches = @q.result(distinct: true).page(params[:page])
```

Comment:
These should also get ordered explicitly.

Reply:
Done